pep517/_in_process.py

```diff
@@ -29,6 +30,23 @@ def __init__(self, traceback):
         self.traceback = traceback
 
 
+class BackendInvalid(Exception):
+    """Raised if we cannot import the backend"""
+    def __init__(self, message):
+        self.message = message
+
+
+def contained_in(filename, directory):
+    """Test if a file is located within the given directory."""
+    filename = os.path.normcase(os.path.abspath(filename))
+    directory = os.path.normcase(os.path.abspath(directory))
+    return os.path.commonprefix([filename, directory]) == directory
```

Comment:
`commonprefix()` appears to be a purely string-based operation, with no awareness of filesystems. So is `commprefix([a, b]) == b` any different from `a.startswith(b)`?

Reply:
Correct, it's a string-based operation. If we didn't need compatibility with older Python versions, I'd use commonpath (or pathlib). If you know of a Python 2 compatible way of saying "is this filename in this directory", I'd happily use it. But in the absence of anything better, I prefer `os.path.commonprefix` because at least it more clearly indicates the intent.